### PR TITLE
Optics documentation

### PR DESCRIPTION
**Pull request by bcamacho**
### Description

Optics technical overview

### Tested

Images uploaded contain a white background and currently pose no issues with the existing color scheme. There might be an issue with dark mode plugins. Considering this is our first publication regarding Optics, we plan to iterate down the road, at which point we should update graphics to remove background colors.

### Documentation

_The set of community-facing docs have been added to support our Kuneco Community All-Hands Optics announcment_

### CODE REVIEW

File: packages/docs/celo-codebase/protocol/optics.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Optics: OPTimistic Interchain Communication
+
+Optics is a new design for radically cheaper cross-chain communication without header verification. We expect operating Optics to cut 90% of gas costs compared to a traditional header relay. To accomplish this, we took inspiration from optimistic systems (a la Optimistic Roll-ups). Optics features many of the features we prize in an optimistic mechanism, like public verification, low gas fees, broad participation, but has a slightly different security model. 
+
+Optics will form the base layer of a cross-chain communication network that provides fast, cheap communication for all smart contract chains and rollups. It relies only on widely-available cryptographic primitives (unlike header relays), has a latency of a few hours (rather than an ORU’s one week latency), and imposes only about 120,000 gas overhead on message senders.
```
Comment (barbaraliau) [inline packages/docs/celo-codebase/protocol/optics.md:5]: How does this gas amount translate into different chains? Will it always cost this much? What's the current "standard/acceptable" cost?

Note: this doesn't have to be changed/answered for this PR

File: packages/docs/celo-codebase/protocol/optics.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+## Building Intuition for Optics
+
+Optics works something like a notary service. 
+
+
+<img width="1172" alt="optics-notary" src="https://user-images.githubusercontent.com/2653576/115466701-4a3c9880-a1e5-11eb-87e3-ae1cbab49b0b.png">
+
+
+The sending (or “home”) chain produces a series of documents (messages) that needs notarization. A notary (called the “updater”) is contracted to sign it. The notary can produce a fraudulent copy, but they will be punished by having their bond and license publicly revoked. When this happens, everyone relying on the notary learns that the notary is malicious. All the notary's customers can immediately block the notary and prevent any malicious access to their accounts.
```
Comment (barbaraliau) [inline packages/docs/celo-codebase/protocol/optics.md]: Add quotations to "messages" for consistency

File: packages/docs/celo-codebase/protocol/optics.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+Fraud is costly
+All users can learn about fraud
+All users can block a fraudulent message before they are accepted
+
+In other words, rather than using a globally verifiable fraud-proof, Optics relies on local verification by participants. This tradeoff allows Optics to save 90% on gas fees compared to pessimistic relays, while still maintaining a high degree of security.
+
+## Optics Architecture
+
+Optics contains several on-chain and off-chain components. For convenience, we’ll be referring to the Home and Replica as contracts, when in fact they are several contracts working together
```
Comment (barbaraliau) [inline packages/docs/celo-codebase/protocol/optics.md]: Nit: add period at end

File: packages/docs/celo-codebase/protocol/optics.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+Simple |  
+
+The benefit of Optics is that the broadcast channel allows for a single-producer, multi-consumer model. This ensures that 1 accumulator can communicate with any number of receiving chains. It’s also much cheaper than other options, allowing updates and proofs to cost <100k gas and be checked by only 1 signature. With Optics, one-solution fits all meaning that constraints on receiving chains are minimal (1 hash function + 1 signature check). There is no implementation or security difference between Proof of Stake and Proof of Work chains.  There is also many fewer LoC than a Relay, much lower design maintenance overhead, and much less expertise required to maintain and operate.
+
+We’ve been careful to address all concerns with the Optics system and have designed solutions that allow for optimal speed, cost, and security of the network. For example, we rely on fraud publication rather than fraud proofs to improve the speed and cost of sending messages. In this security model, any potential fraud is disincentivized and costly, and all participants will always learn of any potential fraud with plenty of time to mitigate harm.
+
+
```
Comment (barbaraliau) [inline packages/docs/celo-codebase/protocol/optics.md:175]: Are there links to the code implementation? If not, we should add them here

**Review by barbaraliau — COMMENTED** [review]
Looks good. Couple small comments. Also, I believe we need to add this page to summary.md @critesjosh correct me if wrong.

File: packages/docs/celo-codebase/protocol/optics.md (hunk 1 above)
Comment (bcamacho) [reply packages/docs/celo-codebase/protocol/optics.md:5]: Good question! The efficiency of Optics provides an estimated 90% savings compared to other bridging methods. We can't control gas costs for other chains as they differ with their network costs. @prestwich would be able to address this in more depth if needed.

File: packages/docs/celo-codebase/protocol/optics.md (hunk 4 above)
Comment (bcamacho) [reply packages/docs/celo-codebase/protocol/optics.md:175]: Not yet. We plan to iterate to include examples in the near future.

**Review by critesjosh — APPROVED** [review]
looks good!

**Pull request merged**